code/measures/bow_similarity.py

```diff
@@ -0,0 +1,103 @@
+import nml_parseutil
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+import datetime
+from nltk.stem import PorterStemmer 
+from nltk.tokenize import word_tokenize
+from nltk.corpus import stopwords
+
+company_article_map = nml_parseutil.create_article_map("2001_sample_10M.nml")
+
+stop_words = set(stopwords.words('english'))
+ps = PorterStemmer()
+
+def stem_and_filter(article):
+    """
+    Takes an Article object and tokenizes the article text into a list, then
+    removes stop words from the list and stems all the remaining words. Returns
+    a list of stemmed words in the article.
+    """
+    text = article.article_text
+    tokenized = word_tokenize(text)
+    no_stop_words = [w for w in tokenized if w not in stop_words]
+    stemmed = [ps.stem(w) for w in no_stop_words]
+    return stemmed
+
+def bow_similarity_score(s1, s2):
+    """
+    Returns the bag-of-words similarity score between an article s1 and article s2.
+    Specifically, measures percentage of words in s1 that are also in s2. 
+    s1, s2 must be tokenized and stemmed articles.
+    """
+    return len(set(s1).intersection(s2)) / len(set(s1))
+
+def articles_older_than(article, article_set):
```

Comment:
Articles will all be older than the current one when you calculate the similarity in-stream.